packages/streamr-eventemitter/include/streamr-eventemitter/EventEmitter.hpp

```diff
@@ -0,0 +1,190 @@
+#ifndef STREAMR_EVENTEMITTER_HPP
+#define STREAMR_EVENTEMITTER_HPP
+
+#include <cstddef>
+#include <functional>
+#include <list>
+#include <mutex>
+#include <optional>
+#include <tuple>
+#include <type_traits>
+
+namespace streamr::eventemitter {
+
+template <typename... HandlerArgumentTypes>
+
+// Event class that all events must inherit from.
+// Usage example:
+// struct MyEvent : Event<int, std::string> {};
+struct Event {
+    class Handler {
+    public:
+        using HandlerFunction = std::function<void(HandlerArgumentTypes...)>;
+
+    private:
+        HandlerFunction handlerFunction;
+        size_t id;
+        bool once;
+
+    public:
+        explicit Handler(
```

Comment:
We have to decide what is the naming convention for member variables. Here you use sometimes same name as constructor parameter and sometimes not. One common way is to use m_var style. I used var_ which is used a lot too but I understand that it might look ugly.

Reply:
I discussed with streamr, and the conclusion was that we will use mMyVariable. Need to update this to the .clang-tidy settings